pubsub/google/cloud/pubsub/connection.py

```diff
@@ -201,6 +202,10 @@ def topic_publish(self, topic_path, messages):
         :rtype: list of string
         :returns: list of opaque IDs for published messages.
         """
+        for key, _ in enumerate(messages):
+            if 'data' in messages[key]:
+                messages[key]['data'] = base64.b64encode(
+                    messages[key]['data']).decode('ascii')
```

Comment:
Also the variable named `key` above **is not** a key, it is an index. That is the point of `enumerate`, getting a value in a list and the index of that item.